contracts/eden/src/eden-micro-chain.cpp

```diff
@@ -1823,6 +1824,59 @@ void filter_block(const subchain::eosio_block& block)
    }  // for(trx)
 }  // filter_block
 
+std::vector<subchain::transaction> ship_to_eden_transactions(
+    std::vector<eosio::ship_protocol::transaction_trace>& traces)
+{
+   std::vector<subchain::transaction> transactions;
+
+   for (const auto& transaction_trace : traces)
+   {
+      if (auto* trx_trace =
+              std::get_if<eosio::ship_protocol::transaction_trace_v0>(&transaction_trace))
+      {
+         subchain::transaction transaction{
+             .id = trx_trace->id,
+         };
+
+         for (const auto& action_trace : trx_trace->action_traces)
+         {
+            if (auto* act_trace = std::get_if<eosio::ship_protocol::action_trace_v0>(&action_trace))
+            {
+               std::optional<subchain::creator_action> creatorAction;
+               if (act_trace->creator_action_ordinal.value > 0)
+               {
+                  const auto& creator_action_trace =
+                      std::get<eosio::ship_protocol::action_trace_v0>(
+                          trx_trace->action_traces[act_trace->creator_action_ordinal.value - 1]);
+                  creatorAction = subchain::creator_action{
+                      .seq = act_trace->creator_action_ordinal,
```

Comment:
Use `creator_action_ordinal` to get the creator's trace, then from that, `->receipt.global_sequence`